include/wx/private/jsscriptwrapper.h

```diff
@@ -33,38 +33,42 @@ class wxJSScriptWrapper
     };
 
     wxJSScriptWrapper(const wxString& js, OutputType outputType)
-        : m_escapedCode(js), m_outputType(outputType)
+        : m_outputType(outputType)
     {
         // Adds one escape level.
         const char *charsNeededToBeEscaped = "\\\"\n\r\v\t\b\f";
-        for (
-            size_t pos = m_escapedCode.find_first_of(charsNeededToBeEscaped, 0);
-            pos != wxString::npos;
-            pos = m_escapedCode.find_first_of(charsNeededToBeEscaped, pos)
-        ) {
-            switch (m_escapedCode[pos].GetValue())
+        m_escapedCode.reserve(js.size());
+        for (wxString::const_iterator it = js.begin(); it != js.end(); ++it)
```

Comment:
I think the performance gain comes from not calling `insert()` rather than not using `find_first_of()`, doesn't it? I.e. it might be still advantageous to keep the old loop. We could add a small benchmark to `tests/benchmarks` to check this.

Reply:
P.S. This commit could/should probably be backported to 3.2 too.

Reply:
Yes the improvement is probably mainly due to the `insert()` calls. But I found the structure easier with a simple loop over the whole string.
And yes this should be backported to 3.2 as it was quite significant difference in my observation. I was trying to send around 5MB of JSON data to my javascript via `RunScriptAsync()` and it took over 90 seconds in debug mode on a M1 mac. With the new loop it's negligible.